### PR TITLE
Added FNF Sprite Support Extension

### PR DESCRIPTION
**Pull request by Aness6040**

### CODE REVIEW

File: extensions/Aness6040/Aness_FNF_Sprites_Support.js [hunk 1]
```diff
@@ -0,0 +1,733 @@
...
+                defaultValue: true,
+              },
+              flipY: {
+                type: Scratch.ArgumentType.BOOLEAN,
+                defaultValue: false,
+              },
+            },
+          },
+          {
+            opcode: 'booleanMenu',
```
Comment (LilyMakesThings) [inline extensions/Aness6040/Aness_FNF_Sprites_Support.js:257]: As opposed to having a separate block for this (that's already duplicated in other extensions), you could let the boolean menu accept reporters, remove (or hide) this block and use the menu as opposed to empty boolean inputs.

File: extensions/Aness6040/Aness_FNF_Sprites_Support.js [hunk 2]
```diff
@@ -0,0 +1,733 @@
...
+          },
+        ],
+        menus: {
+          fileType: ['png', 'xml'],
+          boolean: ['true', 'false'],
+        }
+      };
+    }
+  
+    hello() {
```
Comment (LilyMakesThings) [inline extensions/Aness6040/Aness_FNF_Sprites_Support.js:277]: This function is from the docs, remove it

**Review by LilyMakesThings — CHANGES_REQUESTED** [review]
I'm not entirely sure how to test this extension and the comments are in a different language, so here are just a few notes.

**Comment by Aness6040** [comment]
> I'm not entirely sure how to test this extension and the comments are in a different language, so here are just a few notes.

It's French

File: extensions/Aness6040/Aness_FNF_Sprites_Support.js (hunk 1 above)
Comment (Aness6040) [reply extensions/Aness6040/Aness_FNF_Sprites_Support.js:257]: > As opposed to having a separate block for this (that's already duplicated in other extensions), you could let the boolean menu accept reporters, remove (or hide) this block and use the menu as opposed to empty boolean inputs.

i didnt understand

**Comment by Aness6040** [comment]
hmm

File: extensions/Aness6040/Aness_FNF_Sprites_Support.js [hunk 3]
```diff
@@ -0,0 +1,733 @@
...
+
+    convertSize(args) {
+      const size = args.size;
+      // Ajoutez ici la logique de conversion de HaxeFlixel à Scratch
+      const scratchSize = size * 100; // Exemple de conversion, adaptez selon vos besoins
+      return scratchSize;
+    }
+
+    getFlippedImageURI(args) {
+      const image = new Image();
```
[the remark this answers is not in the document]
Comment (GarboMuffin) [reply extensions/Aness6040/Aness_FNF_Sprites_Support.js:695]: i dont think these can just be ignored, that's an unrestricted fetch

**Comment by GarboMuffin** [comment]
Please try to convert comments etc. to English if possible -- that's what I can read and what most people here can understand. Remember that we will have to maintain your code indefinitely even if you leave so being able to read the comments would be nice

If possible please also provide an example project here; this seems like a rather niche thing

**Comment by Aness6040** [comment]
> Please try to convert comments etc. to English if possible -- that's what I can read and what most people here can understand. Remember that we will have to maintain your code indefinitely even if you leave so being able to read the comments would be nice
> 
> If possible please also provide an example project here; this seems like a rather niche thing

Sorry I couldn't translate the comments because there were too many but it's in French if you need help. On the other hand, I made an example project https://studio.penguinmod.com/?turbo#5286669692106 sorry if it's not on turbowarp but I'm too lazy to do it on share.turbowarp.org

**Comment by gomiigg** [comment]
This is really unrelated but this has to be the FUNNIEST PR name I've ever seen because you could just call it an XML Spritesheet extension or something more general. But the fact you have to specify that they're for FNF sprites is really funny

**Comment by yuri-kiss** [comment]
Closed requested by anness.
Co-Close: @Aness6040

**Pull request closed without merging**